src/libraries/System.DirectoryServices/src/System.DirectoryServices.csproj

```diff
@@ -135,6 +137,8 @@
     <Compile Include="Interop\NativeMethods.cs" />
     <Compile Include="Interop\SafeNativeMethods.cs" />
     <Compile Include="Interop\UnsafeNativeMethods.cs" />
+    <Compile Include="$(CoreLibSharedDir)System\ObsoleteAttribute.cs" Link="System\ObsoleteAttribute.cs" />
```

Comment:
The local definition of the attribute is only needed for `netcoreapp2.0-Windows_NT` build

Reply:
Should we make this a common pattern like we did for PlatformAttributes, where we define a property, like `IncludeInternalObsoleAttributes` true and then we have the right conditions to just include the compile asset on downlevel targets?